src/persistence.rs

```diff
@@ -493,11 +473,12 @@ mod tests {
             key: KeyTree::random(&mut rng).0,
         };
         {
-            let (mut atomic_store, mut storage, mut assets) =
+            let (mut atomic_store, mut storage, mut assets, mut transactions) =
```

Comment:
I don't like this pattern. We don't _need_ to create all the various resource types in order to create an `AtomicKeystoreStorage` and an associated `AtomicStore`. Doing so couples these tests to the behavior of other resources, and forces you to modify this seemingly unrelated test code whenever you add a new resource type.

I changed this in #231 already though, so feel free to leave as is and I'll address in that PR when I fix merge conflicts. Unless you can think of a reason why it should be this way, then just let me know.

Reply:
Don't mind this changing.  I'll leave it for your PR though.